editor/src/messages/defer/defer_message_handler.rs

```diff
@@ -0,0 +1,36 @@
+use crate::messages::prelude::*;
+
+#[derive(Debug, Default, ExtractField)]
+pub struct DeferMessageHandler {
+	after_graph_run: Vec<(u64, Message)>,
+	after_viewport_resize: Vec<Message>,
+	current_graph_submission_id: u64,
+}
+
+#[message_handler_data]
+impl MessageHandler<DeferMessage, ()> for DeferMessageHandler {
+	fn process_message(&mut self, message: DeferMessage, responses: &mut VecDeque<Message>, _: ()) {
+		match message {
+			DeferMessage::AfterGraphRun { mut messages } => {
+				self.after_graph_run.extend(messages.drain(..).map(|m| (self.current_graph_submission_id, m)));
+			}
+			DeferMessage::AfterViewportReady { messages } => {
```

Comment:
It should not be possible to add messages to this buffer if there are currently bounds. An if statement in here would be a good prevention.